compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -795,7 +795,8 @@ object PatternMatcher {
         (tpe isRef defn.IntClass) ||
         (tpe isRef defn.ByteClass) ||
         (tpe isRef defn.ShortClass) ||
-        (tpe isRef defn.CharClass)
+        (tpe isRef defn.CharClass) ||
+        (tpe isRef defn.StringClass)
```

Comment:
Scalac does not emit switch for `String`. Is it a limitation we can lift? Java supports switch for `String` since JDK 7

Reply:
https://docs.oracle.com/javase/specs/jls/se7/html/jls-14.html#jls-14.11

Reply:
Is support for switching on strings actually implemented somewhere in the compiler? The Java support is implemented by desugaring them into two switch statements on integers, there is no bytecode-level support for switching on strings.

Reply:
We currenly do exactly the same as in scalac. We desugar it into if/then/else